common/rfb/CSecurityTLS.cxx

```diff
@@ -203,21 +203,46 @@ bool CSecurityTLS::processMsg()
 void CSecurityTLS::setParam()
 {
   static const char kx_anon_priority[] = ":+ANON-ECDH:+ANON-DH";
+#if GNUTLS_VERSION_NUMBER < 0x030603
+  static const char kx_gnutls_priority_fallback[] = "NORMAL";
+#endif
 
   int ret;
   char *prio;
   const char *err;
 
-  prio = (char*)malloc(strlen(Security::GnuTLSPriority) +
-                       strlen(kx_anon_priority) + 1);
+  if (strcmp(Security::GnuTLSPriority, "") != 0)
+    prio = (char*)malloc(strlen(Security::GnuTLSPriority) +
+                         strlen(kx_anon_priority) + 1);
+  else
+#if GNUTLS_VERSION_NUMBER >= 0x030603
+    prio = (char*)malloc(strlen(kx_anon_priority) + 1);
+#else
+    prio = (char*)malloc(strlen(kx_gnutls_priority_fallback) +
+                         strlen(kx_anon_priority) + 1);
+#endif
+
   if (prio == NULL)
     throw AuthFailureException("Not enough memory for GnuTLS priority string");
 
-  strcpy(prio, Security::GnuTLSPriority);
+  if (strcmp(Security::GnuTLSPriority, "") != 0)
+    strcpy(prio, Security::GnuTLSPriority);
+#if GNUTLS_VERSION_NUMBER < 0x030603
+  else
+    strcpy(prio, Security::kx_gnutls_priority_fallback);
+#endif
+
   if (anon)
     strcat(prio, kx_anon_priority);
```

Comment:
`prio` might be uninitialized here (when new GnuTLS and default prio)

Reply:
I don't see it. In case of default prio, it initializes `prio` for both new and old GnuTLS.

Reply:
Ok, I finally understand what you mean. This one should be fixed.